src/main/java/com/ocient/cli/extract/ExtractConfiguration.java

```diff
@@ -31,6 +33,8 @@ public ExtractConfiguration(final Properties properties)
             compression = Compression.valueOf(config.getString(COMPRESSION, DEFAULT_COMPRESSION).toUpperCase());            
             maxRowsPerFile = config.getInteger(MAX_ROWS_PER_FILE, DEFAULT_MAX_ROWS_PER_FILE);
             skipHeader = config.getBoolean(SKIP_HEADER, DEFAULT_SKIP_HEADER);
+            // This Charset.forName will throw illegalArgumentException if name is not valid.
+            encoding = (config.getString(ENCODING) != null) ? Charset.forName(config.getString(ENCODING)) : DEFAULT_ENCODING;
```

Comment:
Might be clearer to use `Optional`? I mostly like this because it avoids calling `config.getString(ENCODING)` multiple times.

```suggestion
            encoding = Optional.ofNullable(config.getString(ENCODING)).map(Charset::forName).orElse(DEFAULT_ENCODING);
```

Reply:
ok, changed